Source/SwiftLintFramework/Protocols/Rule.swift

```diff
@@ -32,6 +32,12 @@ public protocol ConfigurableRule: Rule {
     func isEqualTo(rule: ConfigurableRule) -> Bool
 }
 
+public protocol ViolationLevelRule: ConfigurableRule {
+    var warning: RuleParameter<Int> { get set }
+    var error: RuleParameter<Int> { get set }
+}
+
+@available(*, deprecated=0.5.6, message="Use ConfigurableRule instead.")
```

Comment:
lol, it's cute that you want to deprecate this "properly", but you should really just remove `ParameterizedRule` entirely.

We've removed things without prior deprecation notice in SwiftLint before, and I'm comfortable continuing to do that freely until SwiftLint hits 1.0.

A cursory search of GitHub seems to indicate that no one outside this project is using the protocol: https://github.com/search?l=swift&q=ParameterizedRule&ref=simplesearch&type=Code&utf8=%E2%9C%93